### ISSUE TITLE
Collapse long comments more predicably

### ISSUE THREAD

**Issue by everdred** [thunder-app/thunder#480]
**Is your feature request related to a problem? Please describe.**
Collapsing a very long comment can leave the comment placeholder outside of the viewport, meaning that the user needs to, to continue reading, scroll back up to find their next unread comment.

**Describe the solution you'd like**
A good solution might be auto-scrolling to the right spot to ensure that the placeholder of the newly-collapsed comment appears within the viewport.

**Describe alternatives you've considered**
Maybe instead of collapsing a comment upwards, it could be collapsed "to" the spot the user tapped, meaning it would always shrink both upwards and downwards, instead of only upwards as it does now.

**Additional context**
This is a very long comment. I don't think I'll read the whole thing or the replies.
![1-resize-1](https://github.com/thunder-app/thunder/assets/5367819/c3d2ba26-3ccd-4cad-bf6f-2a5665711245)

Ok, tapped it to hide it. But wait, where was I?
![2-resize-1](https://github.com/thunder-app/thunder/assets/5367819/7607a8ca-faf4-4bf3-a412-f1c41fdd5290)

Scroll up, scroll up... there we go.
![3-resize-1](https://github.com/thunder-app/thunder/assets/5367819/ba3d4433-47c8-4766-b79a-3618c9601bb1)

**Comment by hjiangsu** [comment]
@micahmo does this relate to the PR draft you have with comment collapsing/scrolling? #512

**Comment by micahmo** [comment]
@hjiangsu No, that PR is only for when you're at the very end of the comments. This issue is specifically when you have "Hide Parent Comment on Collapse" selected. Because the top of the parent comment could be off screen, collapsing the contents means you could lose your place, even though everything works as "expected" (i.e., the scroll position doesn't change). I think this could be fixable once we support comment jumping.
